### PR TITLE
Add word counter

### PR DESCRIPTION
**Pull request by Henni**
Fixes the first item of #10 by adding a word counter to the bottom of each document.

cc @jancborchardt @owncloud/designers

### CODE REVIEW

File: js/app/filters/wordCount.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+app.filter('wordCount', function () {
+	'use strict';
+	return function (value) {
+		if (value && (typeof value === 'string')) {
+			return value.trim().split(/\s+/).length;
```
Comment (Raydiation) [inline js/app/filters/wordCount.js]: Simple idea but also counts characters such as * an # which are pretty frequent in markdown

**Comment by Raydiation** [comment]
Looks good otherwise

**Comment by jancborchardt** [comment]
Two stylistic details:

* the word counter is not properly left aligned with the text, it’s a bit too far to the left
* it overlaps with note content if the note is longer than the viewport. Should be pushed down. Or we put it on the right?

**Comment by Henni** [comment]
@jancborchardt The word counter and the editor have exactly the same padding and it fits for me.

**Comment by jancborchardt** [comment]
Hm, seems to only occur with Markdownified notes.

**Comment by jancborchardt** [comment]
Ok, can’t see an issue anymore. :+1: @Henni if you rebase we can merge this :)

File: templates/note.php [hunk 2]
```diff
@@ -1,3 +1,4 @@
 <div id="app-navigation-toggle" class="icon-menu" style="display:none;"></div>
 	<pre editor notes-timeout-change="save()" notes-autofocus ng-class="{saving: isSaving()}">{{note.content}}</pre>
+	<div class="note-meta">{{note.content | wordCount }} words</div>
```
Comment (jancborchardt) [inline templates/note.php]: Forgot: This needs to be translatable. Ideally with the singular form included when there is 1 word. And we do not need to show anything when there are 0 words.

@DeepDiver1975 how can the singular/plural forms be distinguished?

Comment (Henni) [inline templates/note.php]: @jancborchardt Maybe like this?
https://github.com/owncloud/notes/commit/e993536443a13435e567218007115a6425e459b4

**Comment by Raydiation** [comment]
BTW: for l10n: there are solutions but I think it would make sense to hide the counter if there are no words :)

**Comment by Henni** [comment]
I rebased the pull request.
@Raydiation @jancborchardt if you are ok with the translation (especially the plural) this is ready to merge.

**Comment by jancborchardt** [comment]
Seems good! @DeepDiver1975 is that plural form correct?

@Henni one thing: can you squash the commits into one to keep the history clean?

**Comment by Henni** [comment]
@jancborchardt I'll do that.
@DeepDiver1975 which way of the plural form is better? baa1c26 or 9680678

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 which way of the plural form is better? baa1c26 or 9680678

Both looks wrong to me - you are expecting it to be either singular or plural - but there are languages which have more then 2 forms.

**Comment by DeepDiver1975** [comment]
Or I simply don't get that bloody angular syntax - anyway .... simply call n() and pass in the real count

**Comment by Henni** [comment]
@DeepDiver1975 The problem is that the number of words changes. Sending requests back and forth between the browser and php backend just to check if the translation changed isn't really practical.

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 The problem is that the number of words changes. Sending requests back and forth between the browser and php backend just to check if the translation changed isn't really practical.

there is a js function for translation as well

**Comment by Henni** [comment]
@DeepDiver1975 could you give me a hint what functions to use then as there is no real documentation about it. https://doc.owncloud.org/server/8.2/developer_manual/app/l10n.html

**Comment by Henni** [comment]
I found the js functions. `OC.L10N.translatePlural()` is now used in the `wordCount` filter. I also created an issue to document the javascript localization functions: owncloud/documentation#1479

The PR is now ready for review.
cc @Raydiation @jancborchardt

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 could you give me a hint what functions to use then as there is no real documentation about it. https://doc.owncloud.org/server/8.2/developer_manual/app/l10n.html

@Raydiation mind fixing this part of the docs you did write? As you know there are javascript functions for n() and t()

**Comment by DeepDiver1975** [comment]
> I found the js functions. OC.L10N.translatePlural()

You need to use the shortcut aliases n() and t()

**Comment by irgendwie** [comment]
:+1: LGTM

**Comment by jancborchardt** [comment]
LGTM, RTFM, GTFO :+1:

**Pull request merged**